tests/MongoDB.Driver.Tests/PipelineStageDefinitionBuilderTests.cs

```diff
@@ -103,10 +107,229 @@ public void ChangeStream_should_return_the_expected_result_when_options_isNull()
+
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .Match(new BsonDocument("$expr",
+                    new BsonDocument("$and", new BsonArray
+                    {
+                        new BsonDocument("$eq", new BsonArray { "$stock_item", "$$order_item" }),
+                        new BsonDocument("$gte", new BsonArray { "$instock", "$$order_qty" })
+                    })))
+                .Project(Builders<BsonDocument>.Projection
+                    .Exclude("_id")
+                    .Exclude("stock_item"));
+            
+            var result = PipelineStageDefinitionBuilder.Lookup<BsonDocument,BsonDocument, BsonDocument, IEnumerable<BsonDocument>, BsonDocument>
+                (
+                    warehousesCollection,
+                    new BsonDocument
+                    {
+                        { "order_item", "$item" },
+                        { "order_qty", "$ordered" }
+                    },
+                    pipeline,
+                    new StringFieldDefinition<BsonDocument, IEnumerable<BsonDocument>>("stockdata")
+                );
+
+            RenderStage(result).Document.Should().Be(@"
+                {
+                    '$lookup' :
+                    {
+                        'from' : 'warehouses',
+                        'let' :
+                        {
+                            'order_item' : '$item',
+                            'order_qty' : '$ordered'
+                        },
+                        'pipeline' : [
+                        {
+                            '$match' :
+                            { 
+                                '$expr' :
+                                { 
+                                    '$and' : [
+                                        { '$eq' : ['$stock_item', '$$order_item'] },
+                                        { '$gte' : ['$instock', '$$order_qty'] }]
+                                }
+                            }
+                        },
+                        { '$project' : { '_id' : 0, 'stock_item' : 0 } }],
+                        'as' : 'stockdata'
+                    }
+                }");
+        }
+
+        [Fact]
+        public void Lookup_without_optional_let_should_return_the_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateLet);
+
+            var client = new MongoClient(CoreTestConfiguration.ConnectionString.ToString());
+            var warehousesCollection = client.GetDatabase("test").GetCollection<BsonDocument>("warehouses");
+
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>();
+
+            var result = PipelineStageDefinitionBuilder.Lookup<BsonDocument, BsonDocument, BsonDocument, IEnumerable<BsonDocument>, BsonDocument >
+                (
+                    warehousesCollection,
+                    null,
```

Comment:
What are we supposed to do if let is not null but is an empty BsonDocument?

I suppose we just send the empty BsonDocument to the server...

Reply:
Yep, it looks like I can add additional param checking on empty BsonDocument.

Reply:
It looks like empty let documents is considered by the server in the same way as the null value

Reply:
Then we don't need to do anything special.

If the let argument is null, omit the field from the stage. Otherwise include the document, without checking whether it is empty or not.